Stc 10

JKFerland opened the pull request:
Requested changes to Edit functionality are here. Some add/remove functionality was requested to be edited, but will be reflected in STC-11 PR. 

For the pom dependencies, I had merged in STC-15 to help Jason with merge conflicts. Should I now remove files/dependencies related to STC-15? This is similar to the issue with STC-11. I know now this is not good practice, I was just trying to assist Jason.

jdamerow committed:
[STC-7] create project manager and project class

Commit:
First commit! Finally!!!

Added post and get methods to project controller. Added add.html page to
allow creation of projects from admin's and users. Adding a project
redirects the user to the home page. Only authenticated users can create
projects.

Commit:
[STC-11] created removeprojectcontroller, remove.html

Gave users and admins the ability to remove projects. 2 known errors:
-When a project already exists, NonUniqueResultException is thrown.
Tried catching in removeprojectcontroller and projectmanagerimpl,
however the error is still thrown. Unsure of where to catch it
-When a project does not exist, an exception is also thrown. Based off
first issue, I believe I am not catching exceptions correctly.

Commit:
[STC-11] Updated Project Removal method, added project manager page

Updated Project Removal to remove by ID, not name. Added Project manager
page which lists all projects in the database with the Name, Id,
Description, and date added.

Commit:
[STC-11] Edited remove controller + projects manager

Made it so that remove controller does not take in a project object,
rather an ID. Added remove project functionality to project manager.
More changes to come, but committing this before I merge develop.

Commit:
[STC-11] Added changes based on pull request feedback

*Added logger for Project Deletion error exception
*Changed auth/projects/projects to auth/projects

Commit:
Merge remote-tracking branch 'origin/develop' into story/STC-11

Conflicts:
	scriptoocloud/src/main/webapp/WEB-INF/views/layouts/main.html

Commit:
[STC-11] Changed Remove Project Controller + Added Delete Button

Changed Remove controller to take in ID as Parameter variable, so a
button could be used to delete. Button added to listed projects in
project manager to allow deletion on project.

Commit:
[STC-10] Initial commit, added Edit Project Controller

Added Edit project controller, to be used to edit projects on the
project manager.

Commit:
STC-11

Went through history of PR's on STC-7 and STC-11 and implemented
multiple requested changes.

Commit:
[STC 11] Started Exception Work

Commit:
[STC-11] Continued work on nullIDException

Commit:
[STC-13] Added user tracking to Add Project

Commit:
[STC-11]
Return add page on project add, Retrun delete page on project delete.
Changed security context to allow all users to access /auth/** (project
pages).

Commit:
Merge branch 'story/STC-11' into STC-10

Conflicts:
	scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/ProjectManager.java

Commit:
[STC-10]

In progress work of Edit project functionality.

Commit:
[STC-11] Merged Develop into STC-11

Commit:
[STC-11] Removed TODO stub to pass code factor.

Commit:
[STC-11] Implemented formatting changes and fixed exception handling

Commit:
[STC-11] Updated error message for exception.

JKFerland committed:
Delete .project

I see this here now, deleting from PR.

JKFerland committed:
Delete pom.xml

See file here in PR, deleting.

Commit:
Merge remote-tracking branch 'origin/story/STC-11' into STC-10

Commit:
[STC-10] Fixed some indentation

Commit:
[STC-11] Fixed all indentation, re-added pom file, removed the
NullIDException

Commit:
[STC-10] Re-merged STC-11 to include indentation changes, remove the
NullID exception

Commit:
[STC-10]

Commit:
[STC-10] Implemented Edit Functionality

-Projects now have a Last Modified date, to be set on creation and edit.
-Pom.xml is set to match STC-15 (so Jason can merge when needed)
-Projects can have their names and descriptions changed.

Commit:
[STC-11] Fixed autowired dependencies

Removed/Added some autowired dependencies

JKFerland committed:
Delete pom.xml

Commit:
[STC-11] Fix to pom file

We keep having issues with my pom file not being correct (because it
appears that I keep updating it, but I am not). Pushing the pom version
from STC-15 to maintain compatibility with it.

Commit:
[STC-11] Fixing pom file

Commit:
Merge branch 'story/STC-11' of https://github.com/diging/scriptoocloud.git into story/STC-11

Commit:
[STC-11] Corrected pom file so that there are no changes.

Commit:
[STC-10] Merge STC-11 into STC-10

Commit:
[STC-10] Fixed formatting

Commit:
[STC-10] Removed 2 POST methods in EditProjectController

Previous edit implementation had 2 POST requests, corrected this and now
have one GET, one POST method. Also removed private projectID variable.

JKFerland committed:
Delete .project

Commit:
[STC-11]

Commit:
[STC-11] Fixed constructor / reverted pom

JKFerland committed:
Merge branch 'develop' into story/STC-11

Commit:
[STC-11] Removed database location

Commit:
Merge branch 'story/STC-11' of https://github.com/diging/scriptoocloud.git into story/STC-11

Commit:
[STC-11] Added error handling for add projects

Commit:
[STC-10] Added success message to edit project controller

-Implemented requested changes to Edit functionality
-getProject now returns a project, not an optional project
-A success message now shows when projects are edited

diging-jenkins commented:
Can one of the admins verify this patch?

JKFerland committed:
Merge branch 'develop' into STC-10

Commit:
[STC-11] Alert messages added

-Added message on successful project delete/add
-Added user prompt popup to confirm project deletes

jdamerow commented on scriptoocloud/pom.xml:
```diff
@@ -245,11 +249,32 @@
+		    <groupId>org.yaml</groupId>
+		    <artifactId>snakeyaml</artifactId>
+		    <version>1.28</version>
+		</dependency>
```
same comment as before on the other PRs

JKFerland replied:
Merged into stc-10 as stc-10 already had stc-11 merged in the past.

jdamerow reviewed (changes requested):
Please consolidate with STC-11 and have just one PR instead of having the same code in multiple PRs. It's very hard to review and very repetitive.

Commit:
Merge remote-tracking branch 'origin/story/STC-11' into STC-10

Commit:
Merge branch 'STC-10' of https://github.com/diging/scriptoocloud.git
into STC-10

Commit:
[STC-10] Validator Work Started

-Added Validator for Projects
-Merged STC-11 into stc-10, will conduct all further add functionality
changes to this branch
-Added dependencies for validator to pom.xml
-Removed add project controller constructor
-

JKFerland commented:
STC-11 has been merged into STC-10 and all changes for add project functionality will now be present here. 

I started work on the validator and am having issues getting the messages displayed. I am unsure (from the [guide](https://www.boraji.com/spring-4-mvc-form-validation-example-using-validator-interface) you shared with me what change I would need to make. The add project controller will not submit the post request if there is an error (as expected), but the error messages are not showing correctly. Is there a way to tell if it is not hooked up correctly (instead of th: tags I need to use form: tags), or see why the messages are not being sent? I don't think I need to implement all of the Servlet container steps listed in the guide, but maybe there is something I am missing. Should the if statement possibly return something else (if there are errors)?

Also, in a previous PR, you mentioned changes in STC-15 should not be present in the branch/in the pom file. (Reiterating that I did this so Jason could easily access project add functionality, and I know now it is not best practice. Is it possible to un-merge these changes? How should I resolve this?

jdamerow commented on scriptoocloud/pom.xml:
```diff
@@ -244,6 +259,28 @@
+			<groupId>com.github.docker-java</groupId>
+			<artifactId>docker-java-transport-httpclient5</artifactId>
+			<version>3.2.5</version>
+		</dependency>
```
these changes still don't belong to this ticket, do they?

jdamerow commented on scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/AddProjectController.java:
```diff
@@ -0,0 +1,79 @@
+        if(projectImpl.getDescription().isEmpty()) {
+            model.addAttribute("errorMessage", "Please enter a description.");
+            return "/auth/projects/add";
+        }*/
```
remove commented out code

jdamerow commented on scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/EditProjectController.java:
```diff
@@ -0,0 +1,52 @@
+    @RequestMapping(value = "/auth/edit/{id}", method = RequestMethod.GET)
+    public String get(@Valid @ModelAttribute("project") ProjectImpl projectImpl, BindingResult result,
+            Model model, RedirectAttributes redirectAttrs, Principal principal, @PathVariable("id") int id) {
+        projectImpl = (@Valid ProjectImpl) projectManager.getProject(id);
```
remove `@Valid` annotation

jdamerow commented on scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/EditProjectController.java:
```diff
@@ -0,0 +1,52 @@
+    private ProjectManager projectManager;
+    
+    @RequestMapping(value = "/auth/edit/{id}", method = RequestMethod.GET)
+    public String get(@Valid @ModelAttribute("project") ProjectImpl projectImpl, BindingResult result,
```
`projectImpl` shouldn't be passed as argument if it is not needed (which it doesn't seem to since you're just replacing its value in line 35 anyways)

JKFerland replied:
Removed

jdamerow commented on scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/EditProjectController.java:
```diff
@@ -0,0 +1,52 @@
+            model.addAttribute("projectImpl", new ProjectImpl());
+            return "/auth/edit/{id}";
+        }
+        ProjectImpl project = projectImpl;
```
what is this line for?

JKFerland replied:
Removed, it was present before when getProject returned an optional project.

jdamerow commented on scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/EditProjectController.java:
```diff
@@ -0,0 +1,52 @@
+            return "/auth/edit/{id}";
+        }
+        ProjectImpl project = projectImpl;
+        projectManager.updateProject((int) id, project.getName(), project.getDescription());
```
`id` is already an int so no casting needed

jdamerow commented on scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/web/validation/ProjectValidator.java:
```diff
@@ -0,0 +1,39 @@
+            Pattern.CASE_INSENSITIVE);
+      if (!(pattern.matcher(user.getEmail()).matches())) {
+         err.rejectValue("email", "user.email.invalid");
+      }*/
```
remove if not needed

Commit:
[STC-10]

-Removed necessary request mapping variables
-Removed unnecessary request mapping variables

JKFerland commented:
Reopening PR to see if you may assist in seeing why the error messages are not propagating for adding projects.

Commit:
[STC-10] Final Commit

-Validator messages work, do not need messages.properties anymore.
-Validate projects both in add and edit project functionality
-Removed docker dependencies
